src/Query.re

```diff
@@ -82,16 +82,34 @@ let update = (baseQuery, table, decoder, encoder, record, id, conn) => {
     );
   let sql = {j|UPDATE $table SET ? WHERE $table.`id` = ?|j};
   Sql.Promise.mutate(conn, ~sql, ~params?, ())
-  |> Js.Promise.then_((_) => getById(baseQuery, table, decoder, id, conn));
+  |> Js.Promise.then_(((success, _)) =>
+       if (success == 1) {
+         getById(baseQuery, table, decoder, id, conn)
+         |> Js.Promise.then_(res => Js.Promise.resolve(Result.Ok(res)));
+       } else {
+         PimpMySql_Error.NotFound("ERROR: update failed")
+         |> (x => Result.Error(x))
+         |> Js.Promise.resolve;
+       }
+     );
 };
 
-let softCompoundDelete = (baseQuery, table, decoder, id, conn) => {
+let softCompoundDeleteById = (baseQuery, table, decoder, id, conn) => {
   let params = Json.Encode.([|int @@ id|] |> jsonArray |> Params.positional);
   let sql = {j|
     UPDATE $table
     SET $table.`deleted` = 1, $table.`deleted_timestamp` = NOW()
     WHERE $table.`id` = ?
   |j};
   Sql.Promise.mutate(conn, ~sql, ~params?, ())
-  |> Js.Promise.then_((_) => getById(baseQuery, table, decoder, id, conn));
+  |> Js.Promise.then_(((success, _)) =>
+       if (success == 1) {
+         getById(baseQuery, table, decoder, id, conn)
+         |> Js.Promise.then_(res => Js.Promise.resolve(Result.Ok(res)));
+       } else {
+         PimpMySql_Error.NotFound("ERROR: softCompoundDelete failed")
+         |> (x => Result.Error(x))
```

Comment:
Can use `Result.error`
```reason
PimpMySql_Error.NotFound("Error: softCompoundDelete failed")
|> Result.error
|> Js.Promise.resolve
```

Reply:
I did see functions to call constructors in bs-result but I opted to just use the constructor instead.  Why do we have functions to call a constructor?